scipy/stats/tests/test_discrete_basic.py

```diff
@@ -549,3 +549,15 @@ def test_rv_sample():
     rng = np.random.default_rng(98430143469)
     rvs0 = dist.ppf(rng.random(size=100))
     assert_allclose(rvs, rvs0)
+
+def test__pmf_float_input():
+    # gh-21272
+    # test that `rvs()` can be computed when `_pmf` requires float input
+    
+    class rv_exponential(stats.rv_discrete):
+        def _pmf(self, i):
+            return (2/3)*3**(1 - i)
+    
+    rv = rv_exponential(a=0.0, b=float('inf'))
+    rvs = rv.rvs() # should not crash due to integer input to `_pmf`
+    assert_allclose(rvs, 0)
```

Comment:
I don't know how much this matters but I have just realised that I mistranslated this from the original sympy test. It should be `a  = 1.0`. With `a = 0.0` we get `pmf(0) = 2`. It seems that the implementation happily accepts that and just always returns `0` when sampling. With `a = 1.0` it should generate a range of values like `1,2,...`.

Reply:
I think it's OK because `_pmf` still gets called with integer arguments and raises the error.